spec/front_desk_spec.rb

```diff
@@ -0,0 +1,398 @@
+require_relative "spec_helper"
+
+module Hotel
+  describe "class FrontDesk" do
+    def build_front_desk
+      return Hotel::FrontDesk.new
+    end
+
+    describe "all arrays" do
+      before do
+        @concierge = build_front_desk
+      end
+
+      it "will say that all_rooms is an array" do
+        expect(@concierge.all_rooms_array).must_be_kind_of Array
+      end
+
+      it "will say all_rooms_array has a length of 20" do
+        expect(@concierge.all_rooms_array.length).must_equal 20
+      end
+
+      it "will say element at all_rooms index 0 is 1" do
+        expect(@concierge.all_rooms_array[0]).must_equal 1
+      end
+      it "will say element at all_rooms index 19 is 20" do
+        expect(@concierge.all_rooms_array[19]).must_equal 20
+      end
+
+      it "will say reservations_record is an array" do
+        expect(@concierge.reservations_record).must_be_kind_of Array
+      end
+
+      it "will say block_reservations_only is an array " do
+        expect(@concierge.block_reservations_only).must_be_kind_of Array
+      end
+
+      it "will say reserved_rooms_in_blocks_is an array" do
+        expect(@concierge.reserved_rooms_in_blocks).must_be_kind_of Array
+      end
+
+      it "will say reserved_rooms_in_blocks array length is 0" do
+        expect(@concierge.reserved_rooms_in_blocks.length).must_equal 0
+      end
+
+      it "will say block_reservations_only array length is 0" do
+        expect(@concierge.block_reservations_only.length).must_equal 0
+      end
+
+      it "will say reservations_record array length is 0" do
+        expect(@concierge.reservations_record.length).must_equal 0
+      end
+    end
+
+    describe "make_reservation for block and add_reservation methods" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will describe various attributes of the reservation" do
+        reservation_1 = @concierge.make_reservation(
+          1201,
+          5,
+          1,
+          Date.new(2019, 3, 20),
+          Date.new(2019, 3, 24),
+        )
```

Comment:
Probably would make sense to use keyword arguments here, since you defined them as such in `Reservation`